### DISCUSSION TITLE
Could more programming languages other than C# be added?

### DISCUSSION THREAD

**Discussion by GATETECAT4** [stride3d/stride#1903] [Ideas]
It would be nice if they added more programming languages for the engine, maybe that would attract more people. Although if Stride is closely related to C# and it's going to be difficult to add more, I don't think it's necessary to add more languages, obviously.

I feel that C# might make someone from C++ for example, feel overwhelmed by having to learn another language, but it's not that hard either, it's really more for Strides to have more variety of languages.

**Comment by Jklawreszuk** [comment]
Stride engine is built on top on .NET platform. Technically, This allows game developers to use any CLI language : https://en.wikipedia.org/wiki/List_of_CLI_languages.

Most popular CLI langs:
- C#
- F#
- VisualBasic
- IronPython (sic!)
- ... and many more

Also, One of our contributors @VaclavElias made recently made PR adding support for F#, VB (https://github.com/stride3d/stride/pull/1821), so you can check that out. I'm pretty sure It would be worth noting in our docs (i.e make short article about CLI languages and how you could integrate F# or other CLI with Stride).

Overall, I'd personally say if someone wants to develop games in C++, Stride Engine isn't for them and let them just use any C++ engine/framework. Every programming language/ platform has their own ecosystem of tools and .NET is not exception 😅
I'd also add that my first language I learned was C++ and I think C# syntax,  luckily, is very similar.

**Comment by saint4eva** [comment]
You can use Flax engine
